### PR TITLE
doc: minor updates to OOB migration

### PR DESCRIPTION
**Pull request by unknwon**
Improve examples and add a missing entry in YAML.

## Test plan

Docs update

### CODE REVIEW

File: doc/dev/background-information/oobmigrations.md [hunk 1]
```diff
@@ -179,16 +181,21 @@ func (m *migrator) Up(ctx context.Context) (err error) {
 	}
 	defer func() { err = basestore.CloseRows(rows, err) }()
 
+	updates := make(map[int]string) // id -> payload
```
Comment (eseliger) [inline doc/dev/background-information/oobmigrations.md:184]: This increases memory required to complete the migration, should we note that? Also, what’s the difference here?

**Review by eseliger — APPROVED** [review]
Thank you 🙏

File: doc/dev/background-information/oobmigrations.md (hunk 1 above)
Comment (unknwon) [reply doc/dev/background-information/oobmigrations.md:184]: Oops, forgot to include the context: https://sourcegraph.slack.com/archives/C02UC4WUX1Q/p1658728791213419?thread_ts=1658721255.895179&cid=C02UC4WUX1Q

TLDR we can't do update before finishing iterating the rows

Comment (unknwon) [reply doc/dev/background-information/oobmigrations.md:184]: Maybe I'm doing something obviously wrong? 🤔

Comment (eseliger) [reply doc/dev/background-information/oobmigrations.md:184]: Oooh. No I think that makes sense since in a tx we don’t have access to a connection pool and hence cannot write and read from a result set in parallel.. does that make sense? Can you include that context in the doc too?

Comment (unknwon) [reply doc/dev/background-information/oobmigrations.md:187]: Thanks for a scientific explanation 😉 addressed in https://github.com/sourcegraph/sourcegraph/pull/39405/commits/3028e6dae57e4fd441b27e8048b4e9a5bf5cb489
